packages/trackerless-network/src/logic/StreamrNode.ts

```diff
@@ -249,10 +250,10 @@ export class StreamrNode extends EventEmitter<Events> {
             rpcRequestTimeout: 15000,
             dhtJoinTimeout: 60000,
             nodeName: this.config.nodeName + ':layer1'
-        })
+        }) as ILayer1
```

Comment:
Why casting is needed? (also in tests)

Reply:
It is likely because the `DhtNode` implements the `ITransport` interface. Because of this DhtNodes cannot be directly assigned to `ILayer1` without casting. We also don't want ILayer1 to expose the `ITransport`-methods as they are not needed when interacting with Layer1 nodes.